pkg/logs/launchers/integration/launcher.go

```diff
@@ -85,39 +120,100 @@ func (s *Launcher) run() {
 			for _, source := range sources {
 				// TODO: integrations should only be allowed to have one IntegrationType config.
 				if source.Config.Type == config.IntegrationType {
-					logFilePath, err := s.createFile(cfg.IntegrationID)
+					logFile, err := s.createFile(cfg.IntegrationID)
```

Comment:
What happens if: 

- the agent was restarted
- we run `scanInitialFiles` and are already tracking a file for this integration
- the integration gets scheduled and here we try `createFile`?

Will `createFile` fail?
maybe we should explicitly handle this case?

Reply:
Added logic to avoid creating a file when the integration is already tracked. Also added a test for this scenario.